### DISCUSSION TITLE
Support Metric Selectors for Prometheus data sources in dashboard query variables

### DISCUSSION THREAD

**Discussion by joeblubaugh** [grafana/grafana#65881] [Dashboards, Panels & Visualizations]
Grafana supports several query types when creating dashboard variables. For the Prometheus data source plugin, these are:
* Label Names
* Label Values
* Metrics
* Query Result
* Series Query

Label Values supports a metric selector, which allows you to query the values for a label, as long as the series also match the metric selector. This is useful to say something like: "all the values for the service label on HTTP Request metrics"

This isn't possible for Label Names. The datasource calls this endpoint for label names: https://prometheus.io/docs/prometheus/latest/querying/api/#getting-label-names but it passes no parameters. The API supports a `match[]` query parameter, much like the label values endpoint. Adding support for this would permit asking the question: "what label names are there for all HTTP Request metric series?"

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82768. Feel free to continue the discussion around this there. Thank you!
